src/sbvr-api/abstract-sql.ts

```diff
@@ -140,14 +141,32 @@ export const getAndCheckBindValues = async (
 			}
 
 			try {
-				return await AbstractSQLCompiler[engine].dataTypeValidate(value, field);
+				return await validateBindingType(engine, value, field);
 			} catch (err: any) {
 				throw new BadRequestError(`"${fieldName}" ${err.message}`);
 			}
 		}),
 	);
 };
 
+const validateBindingType = async (
+	engine: Engines,
+	$value: any,
+	field: { dataType: string },
```

Comment:
I think it'd make sense to do
```suggestion
	dataType: string,
```
and then pass in `field.datatype` rather than having to create an object via `{ dataType }` just to recurse

Reply:
@Page- Makes sense. I ended up refactoring the whole piece of code above (the content inside `bindings.map`) as I don't think we needed to have the `field` top level construct anymore.